odoo/addons/base/models/res_users.py

```diff
@@ -1395,6 +1474,21 @@ def unlink(self):
 class UsersView(models.Model):
     _inherit = 'res.users'
 
+    user_group_warning = fields.Text(string="User Group Warning", compute="_compute_user_group_warning")
+
+    @api.depends('groups_id')
+    def _compute_user_group_warning(self):
+        self.user_group_warning = False
+        categories = self.env['ir.module.category'].search([])
+        if self._context.get('show_user_group_warning') and len(self) == 1:
+            group_inheritance_warnings = self.groups_id._prepare_warning_for_group_inheritance(self)
+            # categories for which the empty selection value is selected
+            removed_groups_categories = categories - self.groups_id.category_id
```

Comment:
can't we do `self._origin.groups_id.category_id - self.groups_id.category_id` instead ?
(do we really need to give all categories, even the ones that were not selected before ?)